src/platform/switch/switch_input.c

```diff
@@ -89,14 +91,37 @@ static uint8_t map_switch_button_to_sdlmousebutton[SWITCH_NUM_BUTTONS] =
     NO_MAPPING          // SWITCH_PAD_DOWN
 };
 
-static void switch_start_text_input(char *initial_text, int multiline);
+static void switch_start_text_input(void);
 static void switch_rescale_analog(int *x, int *y, int dead);
 static void switch_button_to_sdlkey_event(int switch_button, SDL_Event *event, uint32_t event_type);
 static void switch_button_to_sdlmouse_event(int switch_button, SDL_Event *event, uint32_t event_type);
 
 static void switch_create_and_push_sdlkey_event(uint32_t event_type, SDL_Scancode scan, SDL_Keycode key);
 static void switch_create_key_event_for_direction(int direction, int key_pressed);
 
+void platform_init_callback(void)
+{
+    touch_set_mode(TOUCH_MODE_TOUCHPAD);
+}
+
+void platform_per_frame_callback(void)
+{
+    if (vkbd_requested) {
+        switch_start_text_input();
+        vkbd_requested = 0;
+    }
+    switch_handle_analog_sticks();
+}
+
+void platform_show_virtual_keyboard(const uint8_t *text)
+{
+    vkbd_text = text ? (const char *)text : "";
```

Comment:
Casting an internal string to C-string is not a good idea for non-ascii text.

Reply:
Hummm. We have a `string_from_ascii`, should I add a `string_to_ascii`? There's still the conversion problem though, I don't really know how Vita/Switch handle the encoding.